### ISSUE TITLE
Support `Point` type for Postgres

### ISSUE THREAD

**Issue by kivi** [prisma/prisma#5984]
## Problem

Can geo location of Postgres added to the supported types?
I can only use: `Unsupported("Point")` for that Postgres data type. And this would not create `create` and `update` for the model that contains that type. 

## Suggested solution

Support `Point` for postgres

## Alternatives

How could implement a workaround for the time being? 
Maybe with rawQueries? Or anybody can come up with a smarter solution?

**Comment by janpio** [comment]
Related: https://github.com/prisma/prisma/issues/1798

**Comment by MurzNN** [comment]
Maybe at now we can implement the postgres `point` field type as `String` prisma field type?
So on create/update queries pass the value `point(1.23, 4.56)` as string in model, and converts to raw sql function in some middleware?
And same trick for reading - receive the value as string `"1.23,4.56"` too.

**Comment by benjamintd** [comment]
Also related: https://github.com/prisma/prisma/issues/5683
This would allow supporting points and other postgis types with custom parse/serialize functions defined by the developer.

**Comment by garytube** [comment]
Please add postgis support!

**Comment by VinayaSathyanarayana** [comment]
Any update on this?

**Comment by floelhoeffel** [comment]
Hello everyone 👋

We are kicking off design of GIS support for Prisma!

If you would like to help, please let us learn about your GIS needs!

**Via survey:**
Please fill out our [GIS User Research Survey](https://prisma103696.typeform.com/to/p8xo5o95) 

**Via email:**
Alternatively, just shoot us an an email to [feedback@prisma.io](mailto:feedback@prisma.io).
The questions we have are (all in the context of GIS):
- what is your GIS use case?
- what database do you use?
- what column types do you use?
- what queries do you use?
- what indexes do you use?
- what kind of defaults do you use?

Thank you!

**Comment by MurzNN** [comment]
Looking to a https://github.com/prisma/prisma/issues/2789 - seems some GEO features are starting to develop, so maybe as a first good step implement the "Point" datatype, as the most easiest one?

**Comment by dagual** [comment]
Our team uses geodata mainly to find objects by coordinates. Adding the "Point" type can greatly facilitate our work.

**Comment by MoutasemHabboub** [comment]
Hi , any new update about supporting "Point" type ?

**Comment by anaibol** [comment]
zzz

**Comment by MurzNN** [comment]
So, for now it's possible to use `Unsupported` field type to make the `Point` field type, like this:
```
model Locality {
  id           String                @id @default(cuid())
  name         String?
  point        Unsupported("Point")?
}
```
And write data into it via `$queryRaw` like this:
```
await prisma.$executeRawUnsafe(`
  UPDATE public."Locality"
  SET point = ST_GeomFromText('POINT(1 2)', 4326)
  WHERE id = '123';
`)
```
But still can't find a way to read the value using Prisma Query (not `$queryRaw`) :(

**Comment by Cyb2rgK1ndr3dSnap** [comment]
I don't know if this post can help you too,  https://alizahid.dev/blog/geo-queries-with-prisma

**Comment by NazCodeland** [comment]
@MurzNN this works for the type but then this throws an error ```ts		
const profile = await prisma.profile.create({
    data: {
        name: `someName`,
            userId: 123
            }
        });
```
`.create()` gives the following error `Property 'create' does not exist on type 'ProfileDelegate<RejectOnNotFound | RejectPerOperation | undefined, DefaultArgs>'.ts(2339)
`

**Comment by NazCodeland** [comment]
@MurzNN when I haave this code ```model Profile {
  location Unsupported("Point")

  @@index([location], type: Gist)
}
```

my `seed.ts` file gives me the error 
```
src/server/database/prisma/seed.ts:20:40 - error TS2339: Property 'create' does not exist on type 'ProfileDelegate<RejectOnNotFound | RejectPerOperation | undefined, DefaultArgs>'.
```
model Profile {
  location Unsupported("Point")? // I can get around this by putting the optional `?`

  @@index([location], type: Gist)
}
```

I can get around this by putting the optional `?` but then in my `create` function I don't have access to `location` as a field in the data object

```typescript
		const profile = await prisma.profile.create({
			data: {
				name: `someName`,
				userId: 123
			}
		});
		```

**Comment by NathBabs** [comment]
> @MurzNN when I have this code
> 
> ```prisma
> model Profile {
>   location Unsupported("Point")
> 
>   @@index([location], type: Gist)
> }
> ```
> 
> my `seed.ts` file gives me the error
> 
> ```
> src/server/database/prisma/seed.ts:20:40 - error TS2339: Property 'create' does not exist on type 'ProfileDelegate<RejectOnNotFound | RejectPerOperation | undefined, DefaultArgs>'.
> ```
> 
> I can get around this by putting `?` after ` location Unsupported("Point")`
> 
> ```ts
> model Profile {
>   location Unsupported("Point")?
> 
>   @@index([location], type: Gist)
> }
> ```
> 
> but then in my `create` function I don't have access to `location` as a field in the data object
> 
> ```ts
> const profile = await prisma.profile.create({
>     data: {
>         name: `someName`,
> 	userId: 123,
> 	// location doesn't exist
> 	}
> });
> ```

Hi have you found a fix for this yet ?

**Comment by NazCodeland** [comment]
@NathBabs no. I ended up changing my stack. I ended up using Kysely to interact with Postgres and my Database types are derived from the Postgres database using Kysely-codegen.

**Comment by NathBabs** [comment]
@NazCodeland you mean kysely supports POSTGIS ?

**Comment by NazCodeland** [comment]
@NathBabs The problem I was having was that I wasn't able to seed my Database using Prisma when I had
```ts
model Profile {
  location Unsupported("Point")

  @@index([location], type: Gist)
}
```

Kysely, is Query builder that I am using to interact with my Database(create, update, delete etc.) In other words, I am seeding my database using Kysely instead of Prisma. 

I am able to seed my Database with Kysely without running into the same issue I ran into when seeding with Prisma.

**Comment by rehatkathuria** [comment]
Have there been any updates or progress on the PostGIS support?

**Comment by Andrew-Pynch** [comment]
+1 for this feature

**Comment by Messhias** [comment]
3 years pasted and we're still waiting for the support for this feature.

**Comment by fafournier** [comment]
+1

**Comment by sprutner** [comment]
+1 on this

**Comment by sprutner** [comment]
Worked around this like so

```
model Location {
  id              Int     @id @default(autoincrement())
[...]
  geopoint        Unsupported("point") // This is actually a POINT datatype and we have to manually add it in a migration.
  @@index([geopoint(ops: raw("point_ops"))], type: Gist)
```

**Comment by pknameer** [comment]
Bump!

**Comment by anaibol** [comment]
+1

**Comment by goran-blazin** [comment]
+1

**Comment by anaibol** [comment]
+1

**Comment by bartfluitsma** [comment]
+1 

Any updates from the team, please?

**Comment by hansinator** [comment]
+1

**Comment by RonAlmog** [comment]
+1, please. or suggest any other way to efficiently store lon/lat data in postgres with prisma.

**Comment by Magellol** [comment]
>suggest any other way to efficiently store lon/lat data in postgres with prisma.

On our end, we've used two separate columns. It's not terrible.

**Comment by fulcanelly** [comment]
+ 1

**Comment by DirckM** [comment]
+1

**Comment by shekharmayank** [comment]
+1

**Comment by kiwavi** [comment]
> > suggest any other way to efficiently store lon/lat data in postgres with prisma.
> 
> On our end, we've used two separate columns. It's not terrible.

Problem arises when you require several points, not one.
